### PR TITLE
Add all artifacts/spells/skills to quest widget lists and disable banned

### PR DESCRIPTION
**Pull request by godric3**
Previous code added at most as many items as were in allowed lists, which caused to skip last items if some were banned.

### CODE REVIEW

File: mapeditor/inspector/questwidget.cpp [hunk 1]
```diff
@@ -48,48 +48,51 @@ QuestWidget::QuestWidget(MapController & _controller, CQuest & _sh, QWidget *par
 	}
 	
 	//fill artifacts
-	for(int i = 0; i < controller.map()->allowedArtifact.size(); ++i)
+	for(auto const & artifactPtr : VLC->arth->objects)
```
Comment (kambala-decapitator) [inline mapeditor/inspector/questwidget.cpp:51]: we use `const auto` style

reference to a pointer isn't needed here

File: mapeditor/inspector/questwidget.cpp [hunk 2]
```diff
@@ -48,48 +48,51 @@ QuestWidget::QuestWidget(MapController & _controller, CQuest & _sh, QWidget *par
...
 	{
-		auto * item = new QListWidgetItem(QString::fromStdString(VLC->artifacts()->getByIndex(i)->getNameTranslated()));
-		item->setData(Qt::UserRole, QVariant::fromValue(i));
+		auto artifactIndex = artifactPtr->getIndex();
+		auto * item = new QListWidgetItem(QString::fromStdString(artifactPtr->getNameTranslated()));
+		item->setData(Qt::UserRole, QVariant::fromValue(artifactIndex));
```
Comment (kambala-decapitator) [inline mapeditor/inspector/questwidget.cpp:55]: please create enum for the custom role

File: mapeditor/inspector/questwidget.cpp [hunk 3]
```diff
@@ -48,48 +48,51 @@ QuestWidget::QuestWidget(MapController & _controller, CQuest & _sh, QWidget *par
...
+	for(auto const & skillPtr : VLC->skillh->objects)
 	{
-		auto * item = new QTableWidgetItem(QString::fromStdString(VLC->skills()->getByIndex(i)->getNameTranslated()));
-		item->setData(Qt::UserRole, QVariant::fromValue(i));
+		auto skillIndex = skillPtr->getIndex();
+		auto * item = new QTableWidgetItem(QString::fromStdString(skillPtr->getNameTranslated()));
+		item->setData(Qt::UserRole, QVariant::fromValue(skillIndex));
 		
 		auto * widget = new QComboBox;
 		for(auto & s : NSecondarySkill::levels)
```
Comment (kambala-decapitator) [inline mapeditor/inspector/questwidget.cpp:85]: please also change here to `const auto &`

File: mapeditor/inspector/questwidget.cpp (hunk 2 above)
Comment (godric3) [reply mapeditor/inspector/questwidget.cpp:55]: hmm... data from those items isn't event retrieved, so I will remove it

**Comment by godric3** [comment]
also added missing serialization of spells in Limiter

**Pull request merged**